Mathlib/Algebra/Module/Zlattice.lean

```diff
@@ -409,23 +426,31 @@ theorem Zlattice.FG : AddSubgroup.FG L := by
     exact fg_span (LinearIndependent.setFinite h_lind)
 
 theorem Zlattice.module_finite : Module.Finite ℤ L :=
-  Module.Finite.iff_addGroup_fg.mpr ((AddGroup.fg_iff_addSubgroup_fg L).mpr (FG K hs))
+  Module.Finite.iff_addGroup_fg.mpr ((AddGroup.fg_iff_addSubgroup_fg L).mpr (FG K L))
+
+instance instModuleFinite_Zlattice {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+    [FiniteDimensional ℝ E] [ProperSpace E] (L : AddSubgroup E) [DiscreteTopology L]
+    [IsZlattice ℝ L] : Module.Finite ℤ L := Zlattice.module_finite ℝ L
```

Comment:
Do you really need L to be a lattice for this instance ? Being discrete should be enough, I guess.